### PR TITLE
Improve cooperative matrix analyis to support cases with scf:ForOp

### PR DESCRIPTION
**Pull request by ThomasRaoux**
Make cooperative matrix analysis support cases where a cooperative matrix is passes to/returned by a scf.For operation.

### CODE REVIEW

File: iree/compiler/Conversion/LinalgToSPIRV/test/convert_to_spirv.mlir [hunk 1]
```diff
@@ -66,3 +66,42 @@ module attributes {gpu.container_module, spv.target_env = #spv.target_env<#spv.v
...
+      // CHECK: spv.Branch ^[[BB]](%{{.*}}, %[[R]] : i32, !spv.coopmatrix<16x16xi32, Subgroup>)
+      scf.yield %8 : vector<16x16xi32>
+    }
+    // CHECK: %[[ACCv:.+]] = spv.Load "Function" %[[ACC]] : !spv.coopmatrix<16x16xi32, Subgroup>
+    // CHECK: spv.CooperativeMatrixStoreNV %{{.*}}, %[[ACCv]], %{{.*}}, %{{.*}}
+    vector.transfer_write %5, %arg2[%c0, %c0] : vector<16x16xi32>, memref<4096x4096xi32>
+    return
+  }
+}
+
```
Comment (hanhanW) [inline iree/compiler/Conversion/LinalgToSPIRV/test/convert_to_spirv.mlir:107]: Not sure if this is Github display issue. I think we don't want a blank line in the end?

There should be a newline after `}` but not two newlines.

Comment (ThomasRaoux) [reply iree/compiler/Conversion/LinalgToSPIRV/test/convert_to_spirv.mlir]: Correct, fixed it.

**Comment by ThomasRaoux** [comment]
> I'm not familiar with cooperative matrix lowering at this point and this seems a small change, so I'll leave the approval to Mahesh.
> 
> I can revisit this once I get some time to study it more. Sorry about that. :(

No worries, thanks Hanhan.

**Comment by MaheshRavishankar** [comment]
Could you add some description to the commit as to what is this adding?

**Comment by ThomasRaoux** [comment]
> Could you add some description to the commit as to what is this adding?

Sure, added some description.

**Comment by ThomasRaoux** [comment]
> I still didnt see the update to commit message, but change looks fine.

Ha yes I updated the pull request description. I'll make sure this makes it as part of the squash and merge commit message.

**Pull request merged**